server_plugin/lib/src/completion.dart

```
@@ -315,95 +312,105 @@ class TemplateCompleter {
           request, extractor.dartSnippet);
       final range =
           new ReplacementRange.compute(dartRequest.offset, dartRequest.target);
-      (request as CompletionRequestImpl)
-        ..replacementOffset = range.offset
-        ..replacementLength = range.length;
+      collector
+        ..offset = range.offset
+        ..length = range.length;
 
       dartRequest.libraryElement = template.view.classElement.library;
       final memberContributor = new TypeMemberContributor();
       final inheritedContributor = new InheritedReferenceContributor();
-
-      suggestions
-        ..addAll(
-          inheritedContributor.computeSuggestionsForClass(
+      inheritedContributor
+          .computeSuggestionsForClass(
             template.view.classElement,
             dartRequest,
             skipChildClass: false,
-          ),
-        )
-        ..addAll(await memberContributor.computeSuggestions(dartRequest));
+          )
+          .forEach(collector.addSuggestion);
+      await memberContributor
+          .computeSuggestions(dartRequest)
+          .then((results) => results.forEach(collector.addSuggestion));
 
       if (dartRequest.opType.includeIdentifiers) {
         final varExtractor = new LocalVariablesExtractor();
         target.accept(varExtractor);
         if (varExtractor.variables != null) {
           addLocalVariables(
-            suggestions,
+            collector,
             varExtractor.variables,
             dartRequest.opType,
           );
         }
       }
-    } else if (target is ElementInfo) {
+      return null;
+    }
+    final replacementRangeCalculator = new ReplacementRangeCalculator(request);
+    target.accept(replacementRangeCalculator);
+    collector
+      ..offset = replacementRangeCalculator.offset
+      ..length = replacementRangeCalculator.length;
```

Unless I'm missing something, the `offset` and `length` are already being set on line 315, so this should throw an exception.